HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -1399,77 +1425,296 @@ def self.add_ee_for_vent_fan_power(model, obj_name, frac_lost, is_cfis, pow = 0.
+        infil_program.addLine("  Set #{htg_energy_actuator.name} = 0.0")
+        infil_program.addLine("  Set Qpreheat = #{UnitConversions.convert(f_preheat.average_oa_flow_rate, 'cfm', 'm^3/s').round(4)}")
+        if [HPXML::MechVentTypeERV, HPXML::MechVentTypeHRV].include? f_preheat.fan_type
+          vent_mech_erv_hrv_tot = [f_preheat]
         else
-          sla = get_infiltration_SLA_from_ACH(air_infil.air_leakage, @infil_height, weather)
-          living_ach50 = get_infiltration_ACH50_from_SLA(sla, 0.65, @cfa, @infil_volume)
+          vent_mech_erv_hrv_tot = []
         end
+        infil_program = calculate_fan_loads(model, infil_program, vent_mech_erv_hrv_tot, hrv_erv_effectiveness_map, fan_sens_load_actuator, fan_lat_load_actuator, 'Qpreheat', true)
+
+        infil_program.addLine("  Set #{htg_energy_actuator.name} = (-FanSensToLv) / #{f_preheat.preheating_efficiency_cop}")
       end
+      infil_program.addLine('EndIf')
     end
+    if not vent_mech_precool.empty?
+      infil_program.addLine('If OASupInTemp > ZoneTemp')
+      vent_mech_precool.each_with_index do |f_precool, i|
+        clg_energy_actuator = create_other_equipment_object_and_actuator(model: model, name: "shared mech vent precooling energy #{i}", space: @living_space, frac_lat: 0.0, frac_lost: 1.0, hpxml_fuel_type: f_precool.precooling_fuel, end_use: Constants.ObjectNameMechanicalVentilationPreconditioning)
+        infil_program.addLine("  Set #{clg_energy_actuator.name} = 0.0")
+        infil_program.addLine("  Set Qprecool = #{UnitConversions.convert(f_precool.average_oa_flow_rate, 'cfm', 'm^3/s').round(4)}")
+        if [HPXML::MechVentTypeERV, HPXML::MechVentTypeHRV].include? f_precool.fan_type
+          vent_mech_erv_hrv_tot = [f_precool]
+        else
+          vent_mech_erv_hrv_tot = []
+        end
+        infil_program = calculate_fan_loads(model, infil_program, vent_mech_erv_hrv_tot, hrv_erv_effectiveness_map, fan_sens_load_actuator, fan_lat_load_actuator, 'Qprecool', true)
 
-    # Infiltration for unconditioned spaces
-    apply_infiltration_to_garage(model, weather, living_ach50)
-    apply_infiltration_to_unconditioned_basement(model, weather)
-    apply_infiltration_to_vented_crawlspace(model, weather, vented_crawl)
-    apply_infiltration_to_unvented_crawlspace(model, weather)
-    apply_infiltration_to_vented_attic(model, weather, vented_attic)
-    apply_infiltration_to_unvented_attic(model, weather)
+        infil_program.addLine("  Set #{clg_energy_actuator.name} = (@Max FanTotalToLv FanSensToLv) / #{f_precool.precooling_efficiency_cop}")
+      end
+      infil_program.addLine('EndIf')
+    end
 
-    # Local ventilation
-    range_sch_sensors_map = apply_local_ventilation(model, vent_fans_kitchen, Constants.ObjectNameMechanicalVentilationRangeFan)
-    bath_sch_sensors_map = apply_local_ventilation(model, vent_fans_bath, Constants.ObjectNameMechanicalVentilationBathFan)
+    return infil_program
```

Comment:
Similar to above comment, shouldn't have to `return infil_program`. Comment probably applies to other methods too.